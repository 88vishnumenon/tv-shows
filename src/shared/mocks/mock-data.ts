import { TVShow } from "../../types/types";

export const mockGenereList: string[] = ["DRAMA", "COMEDY", "THRILLERS"];

export const mockShowList: TVShow[] = [
  {
    showName: "Professor T",
    showImage: {
        "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/482/1205757.jpg",
        "original": "https://static.tvmaze.com/uploads/images/original_untouched/482/1205757.jpg"
      },
    ratings: "7.6",
    summary: "",
    runtime: 60,
    genereList: ["DRAMA", "COMEDY", "THRILLERS"],
    id: 0,
    officialSite: "",
    status: "ENDED",
    schedule: { time: "", days: ["Monday"] },
  },
];





