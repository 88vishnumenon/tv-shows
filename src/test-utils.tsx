import { Provider } from "react-redux";
import { RootState, getStoreWithState, store } from "./store/store";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import * as React from "react";
import { TVShow } from "./types/types";

export function renderWithContext(
  element: React.ReactElement,
  state?: RootState
) {
  const store = getStoreWithState(state);
  const utils = render(
    <Provider store={store}>
      <Router>{element}</Router>
    </Provider>
  );
  return { store, ...utils };
}

export function getStateWithItems(
  tvShows: TVShow[]
): RootState {
  const state: RootState = {
    tvshow: {
      showList: tvShows,
      generesList: ["DRAMA"],
      searchString: "",
      error: false,
    },
  };

  return state;
}
