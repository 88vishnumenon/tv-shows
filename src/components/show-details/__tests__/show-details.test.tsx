import { mockShowList } from "../../../shared/mocks/mock-data";
import { getStateWithItems, renderWithContext } from "../../../test-utils";
import ShowDetails from "../show-details";
import * as React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useSearchParams: () => [new URLSearchParams({ ids: "001,002" })],
}));

beforeEach(() => {
  const store = getStateWithItems(mockShowList);
  renderWithContext(<ShowDetails></ShowDetails>, store);
});
describe("render show details component", () => {
  it("snapshot is rendered", () => {
    const { asFragment } = renderWithContext(<ShowDetails></ShowDetails>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("image tag has value", () => {
    const showImage = screen.getByTestId("show-image") as HTMLImageElement;
    expect(showImage.src).toContain(mockShowList[0].showImage.original);
  });
  it("show detail section has values", () => {
    const scheduleElement = screen.getByTestId("schedule-value") as HTMLElement;
    const genereElement = screen.getByTestId("genere-value") as HTMLElement;
    const statusElement = screen.getByTestId("status-value") as HTMLElement;
    expect(scheduleElement.textContent).toContain(
      `${mockShowList[0]?.schedule?.days.join(",")} at 10:00`
    );
    expect(genereElement.textContent).toContain(mockShowList[0]?.genereList?.join(" | "));
    expect(statusElement.textContent).toContain(mockShowList[0].status);
  });

  // it.todo("show details has values")
});
