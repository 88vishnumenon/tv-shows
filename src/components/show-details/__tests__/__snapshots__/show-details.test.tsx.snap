// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render show details component snapshot is rendered 1`] = `
<DocumentFragment>
  <section
    class="detailsWrapper"
    data-testid="details-page"
  >
    <h1
      class="showTitle"
    />
    <section
      class="detailsSection"
    >
      <img
        class="showImage"
        data-testid="show-image"
      />
      <section
        class="showInformaton"
      >
        <div
          class="description"
        />
        <div
          class="showDetails"
        >
          <div
            class="showDetailValues"
          >
            <span
              class="label"
            >
              Schedule:
            </span>
            <span
              class="values"
              data-testid="schedule-value"
            >
              Monday at 10:00
            </span>
          </div>
          <div
            class="showDetailValues"
          >
            <span
              class="label"
            >
              Rating:
            </span>
            <span
              class="values"
            >
              <span
                class="fa fa-star checked"
              />
              <span>
                 3
              </span>
            </span>
          </div>
        </div>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
